File: 0002 - AddTwoNumbers/Solution.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2) {
    struct ListNode* dummyNode = malloc(sizeof(struct ListNode));
    dummyNode->val = 0;
    dummyNode->next = NULL;
    struct ListNode* current = dummyNode;
    int carry = 0;
    while (l1 || l2 || carry){
        int x = (l1) ? l1->val : 0;
        int y = (l2) ? l2->val : 0;
        int sum = x + y + carry;
        carry = sum/10;
        current->next = malloc(sizeof(struct ListNode));
        current->next->val = sum%10;
        current->next->next = NULL;
        current = current->next;
        if (l1){l1=l1->next;}
        if (l2){l2=l2->next;}

    }
    return dummyNode->next;
}

File: 0002 - AddTwoNumbers/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: Optional[ListNode]
        :type l2: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        dummyHead = ListNode()
        current = dummyHead
        carry = 0
        while l1 or l2 or carry:
            x = l1.val if l1 else 0
            y = l2.val if l2 else 0
            sum = x + y + carry
            carry = sum//10
            current.next = ListNode(sum%10)
            current = current.next

            if l1:
                l1 = l1.next
            if l2:
                l2 = l2.next
        return dummyHead.next


        
